### PR TITLE
fixed, calculate glyphIdArrayLength

### PR DESCRIPTION
**Pull request by prepare**
this is better than the old one,
since cmap provide length field,

(I just notice that today.
then the Its easy to find the end of the cmap)

### CODE REVIEW

**Comment by prepare** [comment]
please send me the proper way to rebase it :)

**Comment by prepare** [comment]
@vidstige 

Hello!, how are you.
1.Do you mind if I ask you to change 
NRasterizer to NOpenType?

2.Do you have a plan for a TrueType Instruction VM?

File: NOpenType/Tables/CmapReader.cs [hunk 1]
```diff
@@ -43,32 +47,33 @@ private static CharacterMap ReadCharacterMap(CMapEntry entry, BinaryReader input
 
                 // I want to thank Microsoft for not giving a simple count on the glyphIdArray
                 //var glyphIdArrayLength = (int)((input.BaseStream.Position - tableStart) / sizeof(UInt16));
-                int glyphIdArrayLength = FindGlyphIdArrayLenInBytes(idRangeOffset) / 2;
+                //int glyphIdArrayLength = FindGlyphIdArrayLenInBytes(idRangeOffset) / 2;
```
Comment (vidstige) [inline NOpenType/Tables/CmapReader.cs:50]: remove comment.

File: NOpenType/Tables/CmapReader.cs [hunk 2]
```diff
@@ -43,32 +47,33 @@ private static CharacterMap ReadCharacterMap(CMapEntry entry, BinaryReader input
...
+                int glyphIdArrayLength = (int)(tableEndAt - input.BaseStream.Position) / 2;
```
Comment (vidstige) [inline NOpenType/Tables/CmapReader.cs:51]: should this not be a long? Then you can also avoid the cast.

File: NOpenType/Tables/CmapReader.cs [hunk 3]
```diff
@@ -43,32 +47,33 @@ private static CharacterMap ReadCharacterMap(CMapEntry entry, BinaryReader input
...
+        //        if (off > max)
+        //        {
+        //            max = off;
+        //            foundAt = i;
+        //        }
+        //    }
+        //    //----------------------------
+        //    //2. then offset with current found record
+        //    return max - (foundAt * 2); //*2 = to byte unit 
+        //}
```
Comment (vidstige) [inline NOpenType/Tables/CmapReader.cs:76]: remove commented out code.

**Comment by vidstige** [comment]
Hi,
Thanks for your contribution! :-) I've been a little sick, but hope to clear up soon, thanks for asking. :-) How are you?

Can you clarify? What should I change? I think I've changed the project name to OpenType. Do you mean the github project?

**Comment by prepare** [comment]
for nuget package.

I don't generate it.
my project (HtmlRenderer)
will merge this at source level :)
Its easy for me.

If you think that nuget package is
useful for every one (I think so)
I encourage you to do that. 👍 

cheer!

**Comment by vidstige** [comment]
You rebase as usual.
1. `git fetch`
2. ´git checkout -b my-rebased-branch-name`  # create new branch to avoid force pushing
3. `git rebase upstream/master`  # rebase, this assumes you've got upstream remote setup.

Then you can push the branch as usual. :-)

**Comment by vidstige** [comment]
Ah, cool. Yeah, I had some problems building it. But the nuspec is checked in.. :-) Can you test it out?
1. `cd NRasterizer`
2. `nuget pack NOpenType.nuspec`. This will create a `.nupkg` file.
3. Add folder as NuGet source in Visual Studio so it can find the package
4. Install package in your project to test out! :-)

**Comment by vidstige** [comment]
Superseeded by #29 .

@prepare I still don't get what you mean with the NuGet package? Did my text help you? Could you try the NuGet package?

**Comment by prepare** [comment]
the current version of 
PixelFarm/HtmlRenderer will use
FontRasterizer at source level.
(It is merged into a large one).

no external dependency is one of the requirement 
of this version

**Comment by vidstige** [comment]
I think I understand. Thank you for clarifying!

**Pull request closed without merging**